android/app/src/main/java/org/uwaterloo/subletr/pages/watcardverification/WatcardVerificationPageView.kt

```diff
@@ -0,0 +1,296 @@
+			) {
+				Text(
+					text = stringResource(id = R.string.watcard_verification_txt1),
+					color = secondaryTextColor,
+				)
+				Text(
+					text = stringResource(id = R.string.watcard_verification_txt2),
+					color = secondaryTextColor,
+				)
+			}
+
+			Spacer(modifier = modifier.weight(2.0f))
+
+			SetContent(
+				viewModel = viewModel,
+				watcard = uiState.watcard
+			)
+
+			Spacer(modifier = modifier.weight(8.0f))
+
+			SetSubmitButton(
+				viewModel,
+				uiState.watcard,
+				uiState.submitted,
+				uiState.verified,
+			)
+
+			Spacer(modifier = modifier.weight(3.0f))
+
+			SetCancelButton(
+				uiState.watcard,
+				uiState.submitted,
+				uiState.verified,
+			)
+
+			Spacer(modifier = modifier.weight(8.0f))
+		}
+	}
+}
+
+@Composable
+fun SetContent(viewModel: WatcardVerificationPageViewModel, watcard : String?) {
+	Box(
+		modifier = Modifier
+			.padding(dimensionResource(id = R.dimen.l))
+			.fillMaxWidth()
+			.height(dimensionResource(id = R.dimen.watcard_image_height))
+			.border(
+				width = dimensionResource(id = R.dimen.xxxs),
+				color = subletrBlack,
+				shape = RectangleShape,
+			)
+			.background(Color.Transparent)
+			.clickable {
+				viewModel.updateUiState(
+					WatcardVerificationUiState.Loaded(
+						watcard = "uploaded",
+						submitted = false,
+						verified = false,
+					)
+				)
+			},
+		contentAlignment = Alignment.Center,
+	) {
+		if (watcard == null) {
+			Column(
+				horizontalAlignment = Alignment.CenterHorizontally,
+				verticalArrangement = Arrangement.Center,
+			) {
+				Icon(
+					modifier = Modifier.size(dimensionResource(id = R.dimen.xl)),
+					painter = painterResource(
+						id = R.drawable.add_filled_black_32
+					),
+					contentDescription = stringResource(id = R.string.add_sign),
+				)
+				Text(
+					stringResource(id = R.string.add_watcard),
+					style = MaterialTheme.typography.bodyMedium,
+				)
+			}
+		} else {
+			// TODO: show picture
+		}
+	}
+}
+
+@Composable
+fun SetSubmitButton(
+	viewModel: WatcardVerificationPageViewModel,
+	watcard: String?,
+	submitted: Boolean,
+	verified : Boolean
+) {
+	var active by remember { mutableStateOf<Boolean>(true) }
+
+	PrimaryButton(
+		modifier = Modifier
+			.fillMaxWidth(ELEMENT_WIDTH)
+			.height(dimensionResource(id = R.dimen.xl)),
+		enabled = active,
+
+		onClick = { // TODO: verify
+			if (watcard != null) {
+				viewModel.updateUiState(
+					WatcardVerificationUiState.Loaded(
+						watcard = watcard,
+						submitted = true,
+						verified = false,
+					)
+				)
+
+			}
+		},
+		colors = ButtonDefaults.buttonColors(
+			containerColor = subletrPink,
+			contentColor = textOnSubletrPink,
+			disabledContainerColor = lightGrayButtonColor,
+		)
+	) {
+		if (!submitted) {
+			active = true
+			Text(
+				stringResource(id = R.string.submit)
+			)
+		} else if (submitted && !verified){
+			active = false
+			Text(
+				stringResource(id = R.string.image_uploaded)
```

Comment:
there should be an icon for the checkmark when the image is uploaded - looks bad as a string